chevah/brink
# [#3744] Use PQM to check PR approval and protected branch.

Scope
=====

This update our merge_init build helper to no longer need changes-approved message as long as they are an explicit approve review message.

It also works with  the Github protected branched, to prevent a merge if the branch tests have not passed.


Changes
=======

Add support for the new review approve message API

Old messages are still supported.

Add support for using the GitHub branch protection.


How to try and test the changes
===============================

reviewers: @alibotean @lgheorghiu @brunogola 

Do the normal review process here.
When using the GitHub review comments you no longer need to add the 'changes-approved'  / 'needs-changes' marker.

`needs-review` is still required.

You can try various level of protections at https://github.com/chevah/brink/settings/branches/master

With the branch protection active for `status checks` you will need to always have a green `paver test_review` before doing a PQM ... as without the test_review the status is not updated.

The PQM is running the tests anyway... but it can't run the aggregated coverage tests. 

In the future I want to enforce 100% coverage for the tests code itself, to make sure there are no tests which are not executed.

Once this is merged, we need to release it and then include it in compat, server and other repors... and set up GitHub protected branches for then. chevah/compat already has the protected branch set up
